File: sdks/python/src/boxsdk/models/shared/events.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from boxsdk import utils
from ..shared import event as shared_event


@dataclass_json
@dataclasses.dataclass
class Events:
    r"""Events
    A list of event objects
    """
    
    chunk_size: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('chunk_size') }})
    entries: Optional[list[shared_event.Event]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('entries') }})
    next_stream_position: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('next_stream_position') }})
    

File: sdks/ts/src/sdk/models/shared/events.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { Event } from "./event";



// Events
/** 
 * A list of event objects
**/
export class Events extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=chunk_size" })
  chunkSize?: number;

  @SpeakeasyMetadata({ data: "json, name=entries", elemType: Event })
  entries?: Event[];

  @SpeakeasyMetadata({ data: "json, name=next_stream_position" })
  nextStreamPosition?: string;
}
